build/document-extractor.js

```diff
@@ -308,18 +309,23 @@ function _addSingleSpecialSection($) {
   let id = null;
   let title = null;
   let isH3 = false;
+  let isH4 = false;
```

Comment:
I would love if we could merge the existing `isH3` and the new `isH4` into a `level: 2 | 3 | 4` attribute on the doc? This can then be consumed directly by the `DisplayHeading` component below.

Reply:
Agreed! I think that would be a good idea to have the heading level within `doc` for easy lookups. I didn't add that logic yet, but if you could direct me on how to add fields to `doc`, I'm happy to handle that.

Reply:
@schalkneethling Could you point me in the right direction for consolidating `isH3/isH4` as an attribute on `doc` ? I understand that we should add `level` as field on `doc`, but I wasn't sure where the file that defined `doc` was located in source. I'm quite busy lately so I apologize for my slow pace for this PR.

Reply:
Sorry, I was off since 13 June. Maybe let's leave this as is for now, as the refactoring may break things.